controllers/discovery/discovery.go

```diff
@@ -84,10 +89,10 @@ type sonataFlowServiceCatalog struct {
 }
 
 // NewServiceCatalog returns a new ServiceCatalog configured to resolve kubernetes, knative, and openshift resource addresses.
-func NewServiceCatalog(cli client.Client) ServiceCatalog {
+func NewServiceCatalog(cli client.Client, knServingClient clientservingv1.ServingV1Interface, knEventingClient clienteventingv1.EventingV1Interface) ServiceCatalog {
```

Comment:
We can have a `DiscoveryClient` struct to hold all this and share it here as a pointer. Wdyt?